modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMFieldLocalServiceImpl.java

```diff
@@ -487,8 +488,20 @@ public void updateDDMFormValues(
 			ddmFieldAttribute.setAttributeName(
 				ddmFieldAttributeInfo._attributeName);
 			ddmFieldAttribute.setLanguageId(ddmFieldAttributeInfo._languageId);
-			ddmFieldAttribute.setAttributeValue(
-				ddmFieldAttributeInfo._attributeValue);
+
+			String attributeValue = ddmFieldAttributeInfo._attributeValue;
+
+			DDMFormField ddmFormField = ddmFormFieldsMap.get(
+				ddmFieldAttributeInfo._ddmFieldInfo._fieldName);
+
+			if ((ddmFormField != null) &&
+				GetterUtil.getBoolean(
+					ddmFormField.getProperty("persistReadOnlyValue"))) {
+
+				attributeValue = ddmFieldAttribute.getAttributeValue();
+			}
+
+			ddmFieldAttribute.setAttributeValue(attributeValue);
```

Comment:
@carolmariaabb 

What do you think if we replace with this:
```
DDMFormField ddmFormField = ddmFormFieldsMap.get(
	ddmFieldAttributeInfo._ddmFieldInfo._fieldName);

if ((ddmFormField != null) &&
	GetterUtil.getBoolean(
		ddmFormField.getProperty("persistReadOnlyValue"))) {

	ddmFieldAttribute.setAttributeValue(ddmFieldAttribute.getAttributeValue());
}
else{
	ddmFieldAttribute.setAttributeValue(ddmFieldAttributeInfo._attributeValue);
}
```

Reply:
I think it's kind of weird to do `ddmFieldAttribute.setAttributeValue(ddmFieldAttribute.getAttributeValue());` :/

Reply:
@gabrielwas I changed to:

```
ddmFieldAttribute.setLanguageId(ddmFieldAttributeInfo._languageId);

DDMFormField ddmFormField = ddmFormFieldsMap.get(
	ddmFieldAttributeInfo._ddmFieldInfo._fieldName);

if ((ddmFormField == null) ||
	!GetterUtil.getBoolean(
		ddmFormField.getProperty("persistReadOnlyValue"))) {

	ddmFieldAttribute.setAttributeValue(
		ddmFieldAttributeInfo._attributeValue);
}

_ddmFieldAttributePersistence.update(ddmFieldAttribute);
```

Reply:
oh yeah! I just inline it and didn't notice that it was the same parameter. Sorry, it's late at night for me haha